src/test/java/com/microsoft/sqlserver/jdbc/fmtOnly/ParameterMetaDataTest.java

```diff
@@ -0,0 +1,151 @@
+                "WITH t1(cInt) AS (SELECT 1), t2(cInt) AS (SELECT 2) SELECT * FROM t1 JOIN t2 ON [t1].\"cInt\" = \"t2\".[cInt] WHERE \"t1\".[cInt] = [t2].\"cInt\" + ?",
+                "INSERT INTO " + tableName + "(cInt,cFloat) SELECT 1,1.5 WHERE 1 > ?",
+                "WITH t1(cInt) AS (SELECT 1), t2(cInt) AS (SELECT 2), t3(cInt) AS (SELECT 3) SELECT * FROM t1,t2,t3 WHERE t1.cInt >= ?",
+                "SELECT (1),2,[cInt],\"cFloat\" FROM " + tableName + " WHERE cNvarchar LIKE ?",
+                "WITH t1(cInt) AS (SELECT 1) SELECT * FROM \"t1\"");
+        l.forEach(this::compareFmtAndSp);
+    }
+
+    @Test
+    public void noStoredProcTest() throws SQLException {
+        List<String> l = Arrays.asList("SELECT cMoney FROM " + tableName + " WHERE cMoney > $?",
+                "SELECT cVarchar FROM " + tableName + " WHERE " + tableName + ".[cVarchar] NOT LIKE ?",
+                "SELECT cVarchar FROM " + tableName + " WHERE ? NOT LIKE " + tableName + ".[cVarchar]",
+                "INSERT " + tableName + "(cInt) VALUES((1+?)),(3*(?+3))", "SELECT ?");
+        l.forEach(this::executeFmt);
+    }
+
+    @Test
+    public void exceptionTest() throws SQLException {
+        executeInvalidFmt("SELECT FROM OPENQUERY INVALID TSQL", "R_invalidOpenqueryCall");
+        executeInvalidFmt("INSERT INTO OPENXML INVALID TSQL VALUES (?,?,?)", "R_invalidOpenqueryCall");
+        executeInvalidFmt("WITH INVALID_CTE AS SELECT * FROM FOO", "R_invalidCTEFormat");
+    }
+
+    private void executeFmt(String userSQL) {
+        try (Connection c = DriverManager.getConnection(AbstractTest.connectionString + ";useFmtOnly=true;");
+                PreparedStatement pstmt = c.prepareStatement(userSQL)) {
+            ParameterMetaData pmd = pstmt.getParameterMetaData();
+            for (int i = 1; i <= pmd.getParameterCount(); i++) {
+                pmd.getParameterClassName(i);
+                pmd.getParameterMode(i);
+                pmd.getParameterType(i);
+                pmd.getParameterTypeName(i);
+                pmd.getPrecision(i);
+                pmd.getScale(i);
+            }
+        } catch (SQLException e) {
+            fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
+        }
+    }
+
+    private void executeInvalidFmt(String userSQL, String expectedError) {
+        try (Connection c = DriverManager.getConnection(AbstractTest.connectionString + ";useFmtOnly=true;");
+                PreparedStatement pstmt = c.prepareStatement(userSQL)) {
+            pstmt.getParameterMetaData();
+        } catch (SQLException e) {
+            assertTrue(e.getMessage().matches(TestUtils.formatErrorMsg(expectedError)));
+            return;
+        }
+        fail(TestResource.getResource("R_expectedFailPassed"));
+    }
+
+    private void compareFmtAndSp(String userSQL) {
+        try (Connection c = DriverManager.getConnection(AbstractTest.connectionString);

```

Comment:
Is connectionString is not modified for new connection, use `AbstractTest.getConnection()` API.